sdk/ml/azure-ai-ml/azure/ai/ml/operations/_component_operations.py

```diff
@@ -248,11 +266,27 @@ def _validate(  # pylint: disable=no-self-use
         if isinstance(component, types.FunctionType):
             component = _refine_component(component)
 
-        # local validation only for now
-        # TODO: use remote call to validate the entire component after MFE API is ready
+        # local validation
         result = component._validate(raise_error=raise_on_failure)
+        # remote validation, note that preflight_operation is not available for registry client
+        if not skip_remote_validation and self._preflight_operation:
+            workspace = self._workspace_operations.get()
+            remote_validation_result = self._preflight_operation.begin_validate(
```

Comment:
Can we make this directly return an validation result entity?

Reply:
I'm afraid not. this operation is from another package (as a vendor pacakge in azure-ai-ml now) and has its own usage.

Another choice is to move the casting logic from `merge_with` to this file. I keep it in `_validation` as it will be shared logic among all resource validation.